src/plugin/ipc/file/fileconnection.cpp

```diff
@@ -251,12 +261,24 @@ void FileConnection::refill(bool isRestart)
                 "   Change this function and re-compile, if you want "
                 "different behavior.");
       } else {
-        const char *errMsg =
+        char *errMsg =
```

Comment:
@rohgarg My restart is failing with the assert failure in while calling from above code.  I guess your check for file overwrite should be moved prior to making areFilesEqual call 

static bool areFilesEqual(int fd, int savedFd, size_t size)
{
  long page_size = sysconf(_SC_PAGESIZE);
  const size_t bufSize = 1024 * page_size;
  char *buf1 =(char*)JALLOC_HELPER_MALLOC(bufSize);
  char *buf2 =(char*)JALLOC_HELPER_MALLOC(bufSize);

  off_t offset1 = _real_lseek(fd, 0, SEEK_CUR);
  off_t offset2 = _real_lseek(savedFd, 0, SEEK_CUR);
  JASSERT(_real_lseek(fd, 0, SEEK_SET) == 0) (fd) (JASSERT_ERRNO);
  JASSERT(_real_lseek(savedFd, 0, SEEK_SET) == 0) (savedFd) (JASSERT_ERRNO);

  int readBytes;
  while (size > 0) {
    readBytes = Util::readAll(savedFd, buf1, MIN(bufSize, size));
    JASSERT(readBytes != -1) (JASSERT_ERRNO) .Text("Read Failed");
    if (readBytes == 0) break;
**    JASSERT(Util::readAll(fd, buf2, readBytes) == readBytes);
**    if (memcmp(buf1, buf2, readBytes) != 0) {
      break;
    }
    size -= readBytes;
  }